### ISSUE TITLE
Improve error catching in reproject

### ISSUE THREAD

**Issue by adehecq** [GlacioHack/geoutils#495]
If mixing the coordinates order in `Raster.reproject`, one is left with a very useless rasterio error that took me a long time to track down...
Here is an example of using reproject with bottom and top swapped:
```
import geoutils as gu
import numpy as np
raster = gu.Raster(gu.examples.get_path("everest_landsat_b4"))
# here I intentionally swap top and bottom
dst_bounds = {"left": raster.bounds.left, "right": raster.bounds.right, "bottom": raster.bounds.top, "top": raster.bounds.bottom}
raster.reproject(bounds=dst_bounds)
```
-> raises a "WindowError: Bounds and transform are inconsistent".

We need to catch this kind of error earlier and raise a more useful error.

Note that this kind of error seem to have somehow not caused any issue in the past, but has become apparent following #443.

**Comment by rhugonnet** [comment]
Yes that would be great, also for `crop()`. It was among the improvements suggested in the end of the Docs on "Homogenization" for raising better error messages!

We could write a check function that is used everywhere the same where rasterization to a new georeferenced grid is involved (`reproject`, `rasterize`, `create_mask`, `proximity`), another one where only bounds are changing (`crop`, and I'm not sure there's another similar function?).
